### ISSUE TITLE
Refactor `sirix-rest-api` bundle and extract more common logic between XML and JSON stuff

### ISSUE THREAD

**Issue by JohannesLichtenberger** [sirixdb/sirix#523]

**Comment by redwanulsourav** [comment]
Hi, I would like to work on this issue.

**Comment by JohannesLichtenberger** [comment]
Give it a try :-) the code base is rather small, but let me know if you need assistance.

**Comment by JohannesLichtenberger** [comment]
@redwanulsourav did you work on the issue already?

**Comment by redwanulsourav** [comment]
@JohannesLichtenberger  I am still working on it, I ran into some issues while setting up the server with docker, I think I have an idea to figure it out. Can I have some more time please?

**Comment by JohannesLichtenberger** [comment]
for sure, just went through some tickets/issues to ping the people if they made some progress :+1: 

BTW: You can improve the documentation if you figured it out :-) currently, I'm mostly working on the core and didn't set up keycloak for some time now. That is usually I'm starting it via the standalone shell script ;)

**Comment by redwanulsourav** [comment]
@JohannesLichtenberger I really appreciate the opportunity and the support. I was able to fix the issues I faced while setting up the keycloak and sirix server using docker. I think the setting up the sirixdb part locally could be updated. Specially the part where keycloak configuration process is described.  I also made some changes to `docker-compose.yml`. 

I would love to contribute in the documentation and improve it. :)

Currently, I was able to run the sirix server and login using `admin` account using `pysirix` client.

I still have to work on this issue itself. I might still need some more time to work on it. As I was able to run the server, I am planning to go through the code base and familiarize myself more. I really appreciate your help and support.

**Comment by JohannesLichtenberger** [comment]
Yes, I have to admit, that I simply downloaded Keycloak, then executed the startup shell script and used IntelliJ for starting/debugging the server...

**Comment by JohannesLichtenberger** [comment]
@redwanulsourav guess you won‘t have time to work on the issue?

**Comment by redwanulsourav** [comment]
@JohannesLichtenberger It was on my radar for a few weeks now. Can you please allow me 3 more weeks? I understand I have taken a long time on this, but I should be able to work on this in the next few weeks. I am sorry for taking long time.

**Comment by JohannesLichtenberger** [comment]
@redwanulsourav do you keep on working on this? Would be great

**Comment by redwanulsourav** [comment]
@JohannesLichtenberger Yes, I will work on this and get the PR finished as soon as possible.

**Comment by JohannesLichtenberger** [comment]
@redwanulsourav any news on this?

**Comment by redwanulsourav** [comment]
I am almost done refactoring two more classes. I got stuck on refactoring one function and got distracted after that. I will be creating a PR for those classes by this weekend.

**Comment by V5t5d** [comment]
Hey! Do you need any help?

**Comment by JohannesLichtenberger** [comment]
Sure, we need every helping hand :-)

**Comment by redwanulsourav** [comment]
@V5t5d Any help is appreciated, I haven't been able to finish this. But I have gone through the code base a little bit and will be able to help if you have any questions.

**Comment by omkar-shitole** [comment]
I see that this issue is still open. If so, I would love to start working on it.
I have already started looking into sirixdb repo a few days back.

Please let me know if there are any specific guidelines or considerations I should be aware of as I dive into this.

**Comment by JohannesLichtenberger** [comment]
You can probably also update Vert.x to the newest version and make changes if necessary :-)
